crates/kubelet/src/server.rs

```diff
@@ -83,12 +84,40 @@ where
     let response = match (req.method(), path.as_slice()) {
         (_, path) if path.len() <= 2 => get_ping(),
         (&Method::GET, [_, "containerLogs", namespace, pod, container]) => {
+            let params: std::collections::HashMap<String, String> = req
+                .uri()
+                .query()
+                .map(|v| {
+                    url::form_urlencoded::parse(v.as_bytes())
+                        .into_owned()
+                        .collect()
+                })
+                .unwrap_or_else(std::collections::HashMap::new);
+            let mut tail = None;
+            let mut follow = false;
+            for (key, value) in &params {
+                match key.as_ref() {
+                    "tailLines" => match value.parse::<usize>() {
+                        Ok(n) => tail = Some(n),
+                        Err(e) => {
+                            warn!(
+                                "Unable to parse tailLines query parameter ({}): {:?}",
+                                value, e
+                            );
+                        }
+                    },
+                    "follow" if value == "true" => follow = true,
+                    s => warn!("Unknown query parameter: {}={}", s, value),
```

Comment:
Do we actually want to warn when follow is explicitly set to "false"?

Reply:
Will fix.